File: src/main/java/eu/zavadil/ocr/stats/OkarinaStats.java
package eu.zavadil.ocr.stats;

import eu.zavadil.java.JavaHeapStats;
import eu.zavadil.java.caching.HashCacheStats;
import eu.zavadil.java.queues.SmartQueueProcessorStats;
import lombok.Getter;
import lombok.Setter;

@Getter
public class OkarinaStats {

	private final JavaHeapStats javaHeap = JavaHeapStats.ofCurrent();

	@Setter
	private SmartQueueProcessorStats documentQueue;

	@Setter
	private HashCacheStats templateCache;

	@Setter
	private HashCacheStats folderChain;

}

File: src/ui/src/types/OkarinaStats.ts
import {CacheStats, JavaHeapStats, QueueStats} from "zavadil-ts-common";

export type OkarinaStats = {
	javaHeap: JavaHeapStats;
	documentQueue: QueueStats;
	templateCache: CacheStats;
	folderChain: CacheStats;
}

export type ClientStats = {
	templatesCache: CacheStats;
	templatesPagesCache: CacheStats;
	fragmentTemplatesCache: CacheStats;
	folderChainCache: CacheStats;
}
